File: Ejercicio 7_Primer_Departamental.cpp
#include <iostream>
#include <string>

using namespace std;

struct AutoPE {
    float precio;
    int anio;
};

class AutoPOO {
public:
    float precio;
    int anio;

    void imprimirDatos() {
        cout << "Auto (Clase) -> Modelo: " << anio << " | Precio: $" << precio << endl;
    }
};


struct PersonaPE {
    string nombre, ap, am, genero;
    int edad;
};

class PersonaPOO {
public:
    string nombre, ap, am, genero;
    int edad;

    void mostrarInfo() {
        cout << "Persona (Clase): " << nombre << " " << ap << " " << am << endl;
        cout << "Edad: " << edad << " | Genero: " << genero << endl;
    }
};


int main() {
    
    AutoPE miAutoStruct = {18500.00, 2019}; 
    
    AutoPOO miAutoClase; 
    miAutoClase.precio = 42000.50;
    miAutoClase.anio = 2025;

    PersonaPE sujetoPE = {"Carlos", "Ruiz", "Sosa", "Masculino", 22};
    
    PersonaPOO sujetoPOO;
    sujetoPOO.nombre = "Elena";
    sujetoPOO.ap = "Torres";
    sujetoPOO.am = "Vaca";
    sujetoPOO.genero = "Femenino";
    sujetoPOO.edad = 21;

    cout << "===== DATOS DE AUTOMOVILES =====" << endl;
    cout << "Auto Struct -> Anio: " << miAutoStruct.anio << " | Precio: $" << miAutoStruct.precio << endl;
    miAutoClase.imprimirDatos();

    cout << "\n===== DATOS DE PERSONAS =====" << endl;
    cout << "Persona Struct -> " << sujetoPE.nombre << " " << sujetoPE.ap << " (" << sujetoPE.edad << " anos)" << endl;
    sujetoPOO.mostrarInfo();

    cout << "==============================" << endl;

    return 0;
}

File: Ejercicio 7_Primer_Departamental.py
class Auto:

    def __init__(self, precio, anio):
        self.precio = precio
        self.anio = anio

    def mostrar_datos(self):
        print("Precio:", self.precio)
        print("Año:", self.anio)

class Persona:

    def __init__(self, nombre, ap, am, genero, edad):
        self.nombre = nombre
        self.apellido_paterno = ap
        self.apellido_materno = am
        self.genero = genero
        self.edad = edad

    def mostrar_datos(self):
        print("Nombre completo:", self.nombre, self.apellido_paterno, self.apellido_materno)
        print("Genero:", self.genero)
        print("Edad:", self.edad)

auto1 = Auto(300000, 2023)
persona1 = Persona("Francisco", "Lopez", "Martinez", "Masculino", 20)

print("Datos del Auto:")
auto1.mostrar_datos()

print("\nDatos de la Persona:")
persona1.mostrar_datos()